### ISSUE TITLE
[Edge WebDriver Docs Feedback]

### ISSUE THREAD

**Issue by Bomarc** [MicrosoftDocs/edge-developer#2613]
Lack of installation directions for 'new' VS install. 

I've installed the current VS / download (64 bit) and attempting to get Selenium (using Edge) working.  The directions for a new installation are ... not very good.  

When followed - the result is an install with references to external URLS - that do not offer any help as they are "upgrade" (vs new install).  The installed code refers to an external URL:
        **// In order to run the below test(s), 
        // please follow the instructions from https://docs.microsoft.com/en-us/microsoft-edge/webdriver-chromium
        // to install Microsoft Edge WebDriver.**

Which when followed refers to both Selenium 3 (upgrade) and 4; with a line "in case you need it".  As this is a fresh in stall; the "in case you need it" should be a "determine if you need it by...." [missing info]"

Your "Note" (ref: "This article provides instructions for using the Selenium framework") has the text "...consult the documentation for your framework of choice." ... that is why I'm at this page.  You should have a link to the Selenium/framework/Edge driver installation/configuration.

Your "sample code" is not in the form of a test; and this needs to be fixed.  (Why else would one want to use Selenium?)

When the sample code is used; it fails, no error message is displayed; no debugging steps are provided.  So - unfortunately I have to say "it fails".  I believe that I've installed most of 'it' correctly; but as there is no debugging/configuration/confirmation steps listed; I can't tell you what "it" is that is failing.

Running the code results in two outputs:

1. "Output" window: 

```
'dotnet.exe' (CoreCLR: DefaultDomain): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.1.30\System.Private.CoreLib.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Users\dan\source\repos\Bomarc\SpamcopReporter\SCR\bin\Debug\netcoreapp2.1\SCR.dll'. Symbols loaded.
'dotnet.exe' (CoreCLR: clrhost): Loaded 'C:\Program Files\dotnet\shared\Microsoft.NETCore.App\2.1.30\System.Runtime.dll'. Skipped loading symbols. Module is optimized and the debugger option 'Just My Code' is enabled.
The program '[9400] dotnet.exe: Program Trace' has exited with code 0 (0x0).
The program '[9400] dotnet.exe' has exited with code 0 (0x0).
```

2. The Debug Console windows:

```
C:\Program Files\dotnet\dotnet.exe (process 11828) exited with code 0.
To automatically close the console when debugging stops, enable Tools->Options->Debugging->Automatically close the console when debugging stops.
```

That is it; there are no test results; there is no automated actions occurring on the display; there is no logging (of test cases or test results) visible; there is no indication **_where_** logging is occurring (if it is); there is no debugging steps provided; there is no "you need to this... ??" to see any of the missing information.

In irony: at the bottom of the (support) page is a link to "See also ... 'The Selenium Browser Automation Page"; which requires the configuration to be correct.  And in closing: Why don't you have sample test cases -- rather than just raw code?

AB#44917120

**Comment by bwalderman** [comment]
@Bomarc, thank you for your feedback.

It sounds like the first source of confusion is the choice between Selenium 3 and Selenium 4. Selenium 3 is out of support and all new projects considering Selenium should use **Selenium 4**. Our docs were initially written a few years ago when Selenium 4 was newly released and our intention was to provide additional context to help migrate away from Selenium 3. Now that Selenium 4 has been released for a while, it might be time to remove references to Selenium 3 from our docs to avoid further confusion.

As noted in the docs, you must install both a browser driver (Microsoft Edge WebDriver), and a WebDriver framework (in this case, Selenium). These are separate components. The steps for installing Microsoft Edge WebDriver are listed under https://learn.microsoft.com/en-us/microsoft-edge/webdriver-chromium/#download-microsoft-edge-webdriver.

It looks like you are using the Edge WebDriver template provided with Visual Studio which creates a simple test project. Once Microsoft Edge WebDriver is downloaded and available on the PATH, and the Selenium 4 NuGet package has been added to the project, then the example test that navigates to Bing should complete successfully.

As you've noticed, our docs provide raw code samples and not complete tests. That's because the purpose of our documentation is not to provide a complete how-to for getting started with Selenium, but to assist with the Microsoft Edge-specific portions of the process. This includes obtaining the Edge WebDriver and setting Edge-specific options in code. Selenium is a cross-browser automation library and is also language and test-framework agnostic. Furthermore, Selenium is just one popular implementation of a WebDriver framework, and Edge WebDriver can be used with any. We use Selenium only as an illustrative example and because it is ideal for most user's needs.

**Comment by captainbrosset** [comment]
Fixed by #2624.
